src/XMLexport.cpp

```diff
@@ -648,6 +720,19 @@ bool XMLexport::writeTrigger( TTrigger * pT )
     writeTextElement( "colorTriggerFgColor", pT->mColorTriggerFgColor.name() );
     writeTextElement( "colorTriggerBgColor", pT->mColorTriggerBgColor.name() );
 
+// TODO: The next bit could be revised for a new - not BACKWARD COMPATIBLE form
+// if/when we get 'version' checking enforced in the 'MudletPackage' element:
+//    int elementCount = qMin( pTt->mRegexCodeList.size(), pT->mRegexCodePropertyList.size() ):
+//    writeStartElement( "RegexList" );
+//    writeAttribute( "size", QString::number( elementCount ) );
+//    for( int i = 0; i < elementCount; ++i ) {
+//        writeEmptyElement( "RegexCode" );
+//        writeAttribute( "index", QString::number( i ) );
+//        writeAttribute( "type", QString::number( pT->mRegexCodePropertyList.at(i) ) );
+//        writeAttribute( "data", pT->mRegexCodeList.at(i) );
```

Comment:
This proposal would include both the content and the type in a new form:

    <RegexList size="n">
        <RegexCode index="1" type="<regexp type number (0-6)>" data="what ever the string is for the 1st regexp" />
        <RegexCode index="2" type="<regexp type number>" data="what ever the string is for the 2nd regexp" />
        ....
        <RegexCode index="n" type="<regexp type number>" data="what ever the string is for the nth regexp" />
    </RegexList>

In hindsight the following might be considered more proper:

    <RegexList size="n">
        <Regex index="1" type="<regexp type number (0-6)>" >what ever the string is for the 1st regexp</RegexCode>
        <Regex index="2" type="<regexp type number>" >what ever the string is for the 2nd regexp</RegexCode>
        ....
        <Regex index="n" type="<regexp type number>" >what ever the string is for the nth regexp</RegexCode>
    </RegexList>

This would eliminate the need for `XMLimport::readIntegerList( QList<int> & list )` all together and remove the need to have a `qFatal()` to handle the (vanishingly small chance) if something other than a number was read there - making things better all around I think!